File: server/task-service/client/userDetails.go
package client

import (
	"encoding/json"
	"go.mongodb.org/mongo-driver/bson/primitive"
	"io"
)

type UserDetails struct {
	ID        primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	Email     string             `bson:"email" json:"email"`
	FirstName string             `bson:"first_name" json:"first_name"`
	LastName  string             `bson:"last_name" json:"last_name"`
	Role      string             `bson:"role" json:"role"`
}

type UserIdsRequest struct {
	UserIds []string `json:"userIds"`
}

type UsersDetails []*UserDetails

func (a *UserDetails) ToJSON(w io.Writer) error {
	e := json.NewEncoder(w)
	return e.Encode(a)
}

func (a *UsersDetails) ToJSON(w io.Writer) error {
	e := json.NewEncoder(w)
	return e.Encode(a)
}

File: client/ui-trello/src/app/models/userDetails.ts
export class UserDetails {
  id: string;
  email: string;
  first_name: string;
  last_name: string;
  role: string;

  constructor(id: string, email: string, first_name: string, last_name: string, role: string) {
    this.id = id;
    this.email = email;
    this.first_name = first_name;
    this.last_name = last_name;
    this.role = role;
  }
}
